climada/test/test_cropyield_integr.py

```diff
@@ -0,0 +1,165 @@
+from climada.entity.exposures.crop_production import CropProduction
+from climada.entity import ImpactFuncSet, IFRelativeCropyield
+from climada.engine import Impact
+
+
+INPUT_DIR = os.path.join(DATA_DIR, 'demo')
+FN_STR_DEMO = 'annual_FR_DE_DEMO'
+FILENAME_LU = 'histsoc_landuse-15crops_annual_FR_DE_DEMO_2001_2005.nc'
+FILENAME_MEAN = 'hist_mean_mai-firr_1976-2005_DE_FR.hdf5'
+
+
+class TestIntegr(unittest.TestCase):
+    """Test loading functions from the ISIMIP Agricultural Drought class and
+        computing impact on crop production"""
+    def test_EU(self):
+        """test with demo data containing France and Germany"""
+        bbox = [-5, 42, 16, 55]
+        haz = RelativeCropyield()
+        haz.set_from_single_run(input_dir=INPUT_DIR, yearrange=(2001, 2005), bbox=bbox,
+                                ag_model='lpjml', cl_model='ipsl-cm5a-lr', scenario='historical',
+                                soc='2005soc', co2='co2', crop='whe', irr='noirr',
+                                fn_str_var=FN_STR_DEMO)
+        hist_mean = haz.calc_mean(yearrange_mean=(2001, 2005))
+        haz.set_rel_yield_to_int(hist_mean)
+        haz.centroids.set_region_id()
+
+        exp = CropProduction()
+        exp.set_from_single_run(input_dir=INPUT_DIR, filename=FILENAME_LU, hist_mean=FILENAME_MEAN,
+                                              bbox=bbox, yearrange=(2001, 2005),
+                                              scenario='flexible', unit='t', crop='whe', irr='firr')
+
+        exp.set_to_usd(INPUT_DIR)
+        exp.assign_centroids(haz, threshold=20)
+
+        if_cp = ImpactFuncSet()
+        if_def = IFRelativeCropyield()
+        if_def.set_relativeyield()
+        if_cp.append(if_def)
+        if_cp.check()
+
+        impact = Impact()
+        impact.calc(exp.loc[exp.region_id == 276], if_cp, haz.select(['2002']), save_mat=True)
+
+        exp_manual = exp.value.loc[exp.region_id == 276].values
+        impact_manual = haz.select(event_names=['2002'], reg_id=276).intensity.multiply(exp_manual)
+        dif = (impact_manual - impact.imp_mat).data
+
+        self.assertEqual(haz.tag.haz_type, 'RC')
+        self.assertEqual(haz.size, 5)
+        self.assertEqual(haz.centroids.size, 1092)
+        self.assertAlmostEqual(haz.intensity.mean(), -2.0489097e-08)
+        self.assertAlmostEqual(exp.value.max(), 53074789.755290434)
+        self.assertEqual(exp.latitude.values.size, 1092)
+        self.assertAlmostEqual(exp.value[3], 0.0)
+        self.assertAlmostEqual(exp.value[1077], 405026.6857207429)
+        self.assertAlmostEqual(impact.imp_mat.data[3], -176102.5359452465 )
+        self.assertEqual(len(dif), 0)
+
+    def test_EU_nan(self):
+        """Test whether setting the zeros in exp.value to NaN changes the impact"""
+        bbox=[0, 42, 10, 52]
+        haz = RelativeCropyield()
+        haz.set_from_single_run(input_dir=INPUT_DIR, yearrange=(2001, 2005), bbox=bbox,
```

Comment:
If the `haz.set_from_single_run` and `set_rel_yield_to_int` are expensive in computation time, look if you can avoid computing it separately for each test maybe. Again, please ask @emanuel-schmid for confirmation that this is a good idea.

Reply:
it is not evry expensive especially because we use small test files. i would suggest to leave it as is.

Reply:
If it is very short fine for me!